android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -243,140 +232,43 @@ void dataChannelSend(String reactTag, String data, String type) {
-            for (RTCStats stats : filteredStats) {
-                filteredStatsMap.put(stats.getId(), stats);
-            }
-            RTCStatsReport filteredStatsReport =
-                    new RTCStatsReport((long) rtcStatsReport.getTimestampUs(), filteredStatsMap);
-            promise.resolve(StringUtils.statsToJSON(filteredStatsReport));
-        });
-    }
-
-    // Note: trackIdentifier can differ from the internal stats trackId
-    // trackIdentifier refers to the sender or receiver id
-    @Nullable
-    private RTCStats getTrackStats(String trackIdentifier, Map<String, RTCStats> statsMap) {
-        for (RTCStats stats : statsMap.values()) {
-            if (stats.getType().equals("track") && trackIdentifier.equals(stats.getMembers().get("trackIdentifier"))) {
-                return stats;
+    public void receiverGetStats(String receiverId, Promise promise) {
+        RtpReceiver targetReceiver = null;
+        for (RtpReceiver r : peerConnection.getReceivers()) {
+            if (r.id().equals(receiverId)) {
+                targetReceiver = r;
+                break;
             }
         }
-        return null;
-    }
 
-    @Nullable
-    private RTCStats getStreamStats(String trackId, Map<String, RTCStats> statsMap) {
-        for (RTCStats stats : statsMap.values()) {
-            if (stats.getType().equals("stream")
-                    && Arrays.asList((String[]) stats.getMembers().get("trackIds")).contains(trackId)) {
-                return stats;
-            }
+        if (targetReceiver == null) {
+            Log.w(TAG, "receiverGetStats(): Receiver ID " + receiverId + " not found");
+            promise.resolve(StringUtils.statsToJSON(new RTCStatsReport(0, new HashMap<>())));
+            return;
         }
-        return null;
-    }
 
-    // Note: trackIdentifier can differ from the internal stats trackId
-    // trackIdentifier refers to the sender or receiver id
-    public Set<RTCStats> getExtraStats(String trackIdentifier, Set<Long> ssrcs, Set<String> codecIds,
-            @Nullable String localCandidateId, @Nullable String remoteCandidateId, Map<String, RTCStats> statsMap) {
-        Set<RTCStats> extraStats = new HashSet<>();
-        for (RTCStats stats : statsMap.values()) {
-            switch (stats.getType()) {
-                case "certificate":
-                case "transport":
-                    extraStats.add(stats);
-                    break;
-            }
+        peerConnection.getStats(targetReceiver, rtcStatsReport -> promise.resolve(StringUtils.statsToJSON(rtcStatsReport)));
+    }
 
-            if (stats.getId().equals(localCandidateId) || stats.getId().equals(remoteCandidateId)) {
-                extraStats.add(stats);
-                continue;
+    public void senderGetStats(String senderId, Promise promise) {
+        RtpReceiver targetSender = null;
+        for (RtpReceiver r : peerConnection.getReceivers()) {
```

Comment:
Why do you use `getReceivers` instead of `getSenders`? My app doesn't receive outbound stats if I don't change this line.

Reply:
Ugh, this is my bad. I'll fix it.

Reply:
Thansk!